webextension/js/popup.js

```diff
@@ -121,6 +122,34 @@ const Logic = {
 
   },
 
+  async showAchievementOrContainersListPanel() {
+    // Do we need to show an achievement panel?
+    let showAchievements = false;
+    const achievementsStorage = await browser.storage.local.get({achievements: []});
+    for (const achievement of achievementsStorage.achievements) {
+      console.log(achievement);
+      if (!achievement.done) {
+        showAchievements = true;
+      }
+    }
+    if (showAchievements) {
+      this.showPanel(P_CONTAINERS_ACHIEVEMENT);
+    } else {
+      this.showPanel(P_CONTAINERS_LIST);
+    }
+  },
+
+  async setAchievementDone(achievementName) {
```

Comment:
Need to figure out better logic here.

Reply:
Updated the logic. This is ready to review and spot-check now, @jonathanKingston

Reply:
So done means "read" essentially? It's a little confusing in name between getting the achievement and marking it done.

Reply:
Essentially ... I put "Done" on the button because the user has to actually click it to clear the achievement. (Hopefully after they rate/tweet/share) So "Done" means whatever the user chooses to do: just read it, or read + rate, or read + share, etc.

Reply:
I'll probably keep the name; `setAchievementRead` would imply that the user only has to *see* the achievement for it to go away.